include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -147,20 +172,27 @@ __pattern_swap(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _Rang
 {
     if (__rng1.size() <= __rng2.size())
     {
-        oneapi::dpl::__internal::__ranges::__pattern_walk_n(
-            __tag,
-            oneapi::dpl::__par_backend_hetero::make_wrapped_policy<__swap1_wrapper>(
-                ::std::forward<_ExecutionPolicy>(__exec)),
-            __f, __rng1, __rng2);
-        return __rng1.size();
+        const std::size_t __n = __rng1.size();
+        auto __exec1 = oneapi::dpl::__par_backend_hetero::make_wrapped_policy<__swap1_wrapper>(
+            std::forward<_ExecutionPolicy>(__exec));
+        auto __future = oneapi::dpl::__par_backend_hetero::__parallel_for(
+            _BackendTag{}, std::move(__exec1),
+            unseq_backend::__brick_swap<decltype(__exec1), _Function, std::decay_t<_Range1>, std::decay_t<_Range2>>{
+                __f, __n},
+            __n, __rng1, __rng2);
+        __future.wait(__par_backend_hetero::__deferrable_mode{});
```

Comment:
What a reason why do you use `__future::__deferrable_wait()` or `__future::wait(__par_backend_hetero::__deferrable_mode{})` ?

Reply:
I guess we can just call `.__deferrable_wait();` here?

Reply:
I see Sergey's in agreement and other code is written similarly here. I just switched to directly call the deferrable wait.